R/utils-ui.R

```diff
@@ -196,36 +196,31 @@ message <- function(...) {
     instead of {.fun message}")
 }
 
-warn_for_verbose <- function(verbose = TRUE, env = caller_env()) {
+warn_for_verbose <- function(verbose = TRUE,
+                             env = caller_env(),
+                             user_env = caller_env(2)) {
   # this is about whether `verbose` was present in the **user's** call to the
   # calling function
-  # don't worry about the `verbose = TRUE` default here
+  # don't worry about the `verbose = TRUE` default in warn_for_verbose()
   if (!lifecycle::is_present(verbose) || isTRUE(verbose)) {
     return(invisible())
   }
 
-  called_from <- sys.parent(1)
-  if (called_from == 0) {
-    # called from global env, presumably in a test or during development
-    caller <- "some_googledrive_function"
-  } else {
-    called_as <- sys.call(called_from)
-    if (is.call(called_as) && is.symbol(called_as[[1]])) {
-      caller <- as.character(called_as[[1]])
-    } else {
-      caller <- "some_googledrive_function"
-    }
-  }
   lifecycle::deprecate_warn(
     when = "2.0.0",
-    what = glue("{caller}(verbose)"),
+    what = I(cli::format_inline("The {.arg verbose} argument")),
     details = c(
       "Set `options(googledrive_quiet = TRUE)` to suppress all googledrive messages.",
       "For finer control, use `local_drive_quiet()` or `with_drive_quiet()`.",
       "googledrive's `verbose` argument will be removed in the future."
     ),
+    user_env = user_env,
+    always = identical(env, global_env()),
```

Comment:
I do like to be able to call this after `load_all()` in the global env and see the message.

Reply:
That makes sense, but again if you want to feature it in R packages, I'd think you'd be better off sticking to standard usage.

Reply:
I can redact that part in the book, meaning the specification of `always =`.